### ISSUE TITLE
Allow sharing of type-names between queries.

### ISSUE THREAD

**Issue by csilvers** [Khan/genqlient#48]
We have two different gql queries:
```
	_ = `# @genqlient
	query TestPrep_GetAssessmentItemsByShas($exerciseID: String!, $shas: [String!]!) {
	  assessmentItemsByShas(exerciseId: $exerciseID, shas: $shas) {
	    contentId
	    itemData
	    itemShapeType
	    name
	    sha
	    tagIds
	  }
	}`

	_ = `# @genqlient
	query TestPrep_GetAssessmentItemsForExercise($exerciseID: String!) {
	  exerciseById(id: $exerciseID) {
		assessmentItems {
	      contentId
	      itemData
	      itemShapeType
	      name
	      sha
	      tagIds
	    }
	  }
	}`
```
While the graphql queries differ, the "return" type is the same.  It would be convenient if they could both be the same type in genqlient, rather than two different types with identical shapes.

This is probably related to #12.

**Comment by benjaminjkraft** [comment]
Good idea!  This would need to be opt-in, I think, since it does make those types less stable (if you change one of the queries, say).  But #12 is naturally opt-in, so it would make a lot of sense for them to go together.

**Comment by benjaminjkraft** [comment]
Ah, another option for how to do this would be via fragments; if you define a shared fragment, and spread that into all the queries, then genqlient can notice that and share the type.  This is natural because then the queries actually guarantee that the types match.  But of course it depends on #8.

**Comment by benjaminjkraft** [comment]
Ok, I think #8 (implemented) and/or #12 (in progress) will cover this!
